cloud_blobstore/s3.py

```diff
@@ -404,6 +405,55 @@ def get_size(
                 raise BlobNotFoundError(f"Could not find s3://{bucket}/{key}") from ex
             raise BlobStoreUnknownError(ex)
 
+    def multipart_parallel_upload(
+            self,
+            bucket: str,
+            key: str,
+            src_file_handle: typing.BinaryIO,
+            part_size: int,
+            content_type: str=None,
+            metadata: dict=None,
+            parallelization_factor=8) -> typing.Sequence[dict]:
+        """
+        Upload a file object in parallel.
+        :param bucket:
+        """
+        kwargs = dict()
+        if content_type is not None:
+            kwargs['ContentType'] = content_type
+        if metadata is not None:
+            kwargs['Metadata'] = metadata
+        mpu = self.s3_client.create_multipart_upload(Bucket=bucket, Key=key, **kwargs)
+
+        size = src_file_handle.getbuffer().nbytes
```

Comment:
`getbuffer()` appears to only exist for bytesio.

Reply:
Thanks, I'll make size a parameter